File: Programmers/Level 2/기능개발/Solution.py
def solution(progresses, speeds):
    answer = []

    progresses = progresses[::-1]
    speeds = speeds[::-1]

    while True:
        cnt = 0
        top = progresses[-1]

        while top >= 100 and progresses:
            progresses.pop()
            cnt += 1
            if progresses:
                top = progresses[-1]
        if cnt != 0:
            answer.append(cnt)
            if not progresses:
                break

        for i in range(len(progresses)):
            if progresses[i] < 100:
                progresses[i] += speeds[i]

    return answer

File: Programmers/Level 2/기능개발/Solution.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int index = 0;
    bool run = true;
    
    while(run){
        int count = 0;
        
        for(int i=0;i<progresses.size();i++){
            if(progresses[i]<100)
                progresses[i] += speeds[i];
        }
        
        for(int i=index;i<progresses.size();i++){
            
            if(progresses[i]>=100)
                count++;
            else{
                index = i;
                break;
            }
            
            if(i==progresses.size()-1&&progresses.back()>=100)
                run = false;
        }
        
        if(count>0)
            answer.push_back(count);
    }
    
    return answer;
}